Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadMaterialBaker.cs

```diff
@@ -180,9 +37,16 @@ public void PurgeMaterials(string namePrefix)
     {
       foreach (var entry in materialDict)
       {
-        if (entry.Key.Contains(namePrefix))
+        try
         {
-          materialDict.Remove(entry.Value);
+          if (entry.Key.Contains(namePrefix))
+          {
+            materialDict.Remove(entry.Value);
+          }
+        }
+        catch (Exception e) when (!e.IsFatal())
+        {
+          _logger.LogError(e, "Failed to purge a material from the document."); // TODO: Check with Jedd!
```

Comment:
Would it be useful to have, in lined any more information about the material or document? maybe the `namePrefix`?

```"Failed to purge {materialNamePrefix} from the document"```